### PR TITLE
Fix the Build!!!

### PR DESCRIPTION
**Pull request by korydraughn**
Fixes the build issues as well as adds lots of code that brings the library closer to conformance with the standard.

### CODE REVIEW

**Comment by korydraughn** [comment]
Once this is merged, feel free to delete the other branches.  This PR includes the changes in the std_compliant branch as well as other important changes.  The other branches (copyable_*) have interesting iterator implementations, but don't follow the std and I doubt they will be needed in the future.

**Comment by trel** [comment]
want to squash?

**Comment by korydraughn** [comment]
What do you prefer?  One commit or just a few?
There are a lot of changes in this PR.

**Comment by trel** [comment]
Fine merging as is.

File: unit_tests/src/main.old.cpp [hunk 1]
```diff
@@ -51,6 +51,8 @@ int main(int _argc, char* _argv[])
         rErrMsg_t errors;
         auto* conn = rcConnect(env.rodsHost, env.rodsPort, env.rodsUserName,
                                env.rodsZone, 0, &errors);
+        //auto* conn = rcConnect(env.rodsHost, env.rodsPort, "rods",
```
Comment (trel) [inline unit_tests/src/main.old.cpp]: remove these two lines?

File: unit_tests/src/main.old.cpp [hunk 2]
```diff
@@ -66,18 +68,63 @@ int main(int _argc, char* _argv[])
 
         try
         {
+            std::cout << std::boolalpha;
+
+            //fs::path home = "/tempZone";
```
Comment (trel) [inline unit_tests/src/main.old.cpp]: remove comment

File: unit_tests/src/main.old.cpp [hunk 3]
```diff
@@ -66,18 +68,63 @@ int main(int _argc, char* _argv[])
...
             fs::path home = "/tempZone/home/kory";
-            auto src = home / "col1.d";
-            auto dst = home / "col1.renamed.d";
+            //auto src = home / "col1.d";
+            //auto dst = home / "col1.renamed.d";
```
Comment (trel) [inline unit_tests/src/main.old.cpp]: more comments

File: unit_tests/src/main.old.cpp [hunk 4]
```diff
@@ -66,18 +68,63 @@ int main(int _argc, char* _argv[])
...
+            for (auto&& e : home / "main.renamed.cpp")
+                std::cout << e << '\n';
+            std::cout << '\n';
+
+            std::cout << "is equal : " << (home / "main.cpp" == home / "main.renamed.cpp") << '\n';
+            std::cout << "less than: " << (home / "main.cpp" <  home / "main.renamed.cpp") << '\n';
+            std::cout << "less than: " << (home / "main.renamed.cpp" >= home) << "\n\n";
+
+            //fs::rename(conn, home / "main.cpp", home / "main.renamed.cpp");
+            //fs::rename(conn, home / "col1.renamed.d", home / "test.d");
```
Comment (trel) [inline unit_tests/src/main.old.cpp]: comments

File: unit_tests/src/main.old.cpp [hunk 5]
```diff
@@ -66,18 +68,63 @@ int main(int _argc, char* _argv[])
...
 
-            std::cout << std::boolalpha;
+            std::cout << "\nusing ranged-for recursive_collection_iterator:\n";
+            fs::recursive_collection_iterator rit{conn, "/tempZone/home"};
+            //rit.disable_recursion_pending();
+            //for (const auto& e : fs::recursive_collection_iterator{conn, root})
+            for (const auto& e : rit)
+                std::cout << e << '\n';
+
+            /*
```
Comment (trel) [inline unit_tests/src/main.old.cpp]: commented out code everywhere!

File: unit_tests/src/main.old.cpp (hunk 1 above)
Comment (korydraughn) [reply unit_tests/src/main.old.cpp]: Would changing these values to be placeholders work as well?  For example, <username> and <password>.

Comment (korydraughn) [reply unit_tests/src/main.old.cpp]: Deleted main.old.cpp.

File: unit_tests/src/main.old.cpp (hunk 3 above)
Comment (korydraughn) [reply unit_tests/src/main.old.cpp]: Deleted main.old.cpp.

File: unit_tests/src/main.old.cpp (hunk 2 above)
Comment (korydraughn) [reply unit_tests/src/main.old.cpp]: Deleted main.old.cpp.

File: unit_tests/src/main.old.cpp (hunk 4 above)
Comment (korydraughn) [reply unit_tests/src/main.old.cpp]: Deleted main.old.cpp.

File: unit_tests/src/main.old.cpp (hunk 5 above)
Comment (korydraughn) [reply unit_tests/src/main.old.cpp]: Deleted main.old.cpp.

**Pull request merged**